Source/LinqToDB/Sql/Sql.Strings.cs

```
@@ -286,5 +286,20 @@ protected override SqlExpression TruncateExpression(ISqlExpression value, ISqlEx
 		}
 
 		#endregion
+
+		#region Regex
+
+		[Expression(PN.MySql, "{0} REGEXP {1}", ServerSideOnly = true, IsPredicate = true)]
+		[Expression(PN.SQLite, "{0} REGEXP {1}", ServerSideOnly = true, IsPredicate = true)] // only possible with custom user function, see TableFunctionTests.Regex
+		[Expression(PN.PostgreSQL, "{0} ~ {1}", ServerSideOnly = true, IsPredicate = true)]
+		[Expression(PN.Oracle, "REGEXP_LIKE({0}, {1})", ServerSideOnly = true, IsPredicate = true)]
+		[Expression(PN.SapHana, "{0} REGEXP {1}", ServerSideOnly = true, IsPredicate = true)]
+		[Expression(PN.SqlServer, "dbo.regex({0}, {1})", ServerSideOnly = true, IsPredicate = true)] // supported via https://github.com/infiniteloopltd/SQLServerRegex
```

I would remove this line; sql server should throw an exception, and if consumers want to use regex, they can add the library specifically. Having this will create a false expectation for the consumer, if they look at the failed query and see `dbo.regex()` in the querystring, and wonder why they don't have the `regex` method.

---

I tend to agree, I wouldn't commit core linq2db functions to custom libraries.
What if MS adds Regex support in the future (quite possible) and it's not exactly the same dialect / not 100% compatible with this lib?

If that lib is popular enough to be included in main linq2db lib, I'd put it on a specific static class: `InfiniteLoop.Regex(..)`.